pennylane/compiler/compiler.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Compiler developer functions"""
+
+from typing import List, Optional
+from sys import version_info
+from importlib import reload, metadata
+from collections import defaultdict
+import dataclasses
+
+
+class CompileError(Exception):
+    """Error encountered in the compilation phase."""
+
+
+@dataclasses.dataclass
+class AvailableCompilers:
+    """This contains data of installed PennyLane compiler packages."""
+
+    entrypoints_interface = ("qjit", "context", "ops")
```

Comment:
are these documented anywhere? It seems there are assumptions being made about what's defined, eg. the "ops" interface assumes the compiler has the `jacobian` function implemented. can docs for these interfaces be provided/linked in the "Adding a compiler" section of the module docs?

Reply:
I've added code comments describing them here. :+1: